RemoveData/src/RemoveData.java

```diff
@@ -0,0 +1,134 @@
+import java.util.Date;
+import org.bson.Document;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.result.DeleteResult;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.lt;
+
+public class RemoveData {
+	
+	/**
+	 * Removes all the cancelled orders older than the given date.
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param thresholdDate: to filter documents for deletion.
+	 * @return long: Count of deleted documents..
+	 */
+	public long RemoveAllCancelledOrders(String connectionString, Date thresholdDate){
+		if(connectionString == null || connectionString.isEmpty() || thresholdDate == null)
+		{
+			throw new IllegalArgumentException();
+		}
+		else
+		{
+			MongoClientURI clientUri = new MongoClientURI(connectionString);
+			try(MongoClient client = new MongoClient(clientUri))
+			{
+				MongoDatabase database = client.getDatabase("stores");			
+				MongoCollection<Document> collection = database.getCollection("orders");
+				
+				DeleteResult deletedOrders = collection.deleteMany(and(eq("status","cancelled"),lt("orderPlaced", thresholdDate)));
+				System.out.println("Number of Orders removed: "+deletedOrders.getDeletedCount());
+				return deletedOrders.getDeletedCount();								
```

Comment:
Here we are checking only the number of orders deleted. Meaning that we are not verifying that 5 orders with status "cancelled" are deleted, but only verifying that 5 orders are canceled. Of course it uses the eq to find those satisfy the condition, but maybe we want to check if a specific order( get id number) is deleted? (check if that specific id number is deleted) 
Just a thought

Reply:
The unit test automatically creates 5 dummy records that match a specific condition before deletion, calls remove function with the same condition and verifies if 5 dummy records were deleted. I cannot verify by id unless I hard code 5 records manually and make sure they match the condition.

Reply:
ok got it